ignite/engine/events.py

```diff
@@ -237,7 +237,13 @@ def function_before_backprop(engine):
                 # ...
     """
 
-    pass
+    def __new__(cls, value: str) -> "EventEnum":
+        obj = CallableEventWithFilter.__new__(cls)
+        obj._value_ = value
+        return obj
+
+    def __repr__(self) -> str:
```

Comment:
@sadra-barikbin is this really necesary ?

Reply:
`__repr__` wasn't necessary. I removed it. But `__new__` is.

Reply:
@vfdev-5 , from #2997 turns out it was necessary. Shall I add it as a follow-up or we should ask 3.11 users to update to `3.11.4`?